templates/master/00-master/sno/files/sno-forcedns-fix.yaml

```diff
@@ -0,0 +1,13 @@
+mode: 0755
+path: "/etc/NetworkManager/dispatcher.d/forcedns-rhel9-fix"
+contents:
+  inline: |
+    #!/bin/bash
+    # this script was added as upgrade fix for assisted sno installation were forcedns dispatcher script misses shebang
+    filename="/etc/NetworkManager/dispatcher.d/forcedns"
+    if [ ! -f "$filename" ];
+    then
+      exit
+    fi
+    # in case there is no shebang run forcedns dispatcher script with /bin/bash
+    grep -qxF '#!/bin/bash' "$filename" || /bin/bash "$filename" $1 $2
```

Comment:
Better check that the FIRST line of the file is the shebang:
```
head -n1 "$filename" | grep -qx '#!/bin/bash' || /bin/bash "$filename" $1 $2
```

Reply:
sure? I mean and if it is seconds line by some unknown reason?

Reply:
AFAIK the shebang MUST be the first line of the script always ...